File: backend/src/main/java/org/backend/modules/Review.java
package org.backend.modules;

import org.backend.misc.RoomInfo;
import org.backend.user.UserData;

import java.io.Serializable;
import java.util.Date;
import java.util.Objects;

public class Review implements Serializable {
    UserData userData;
    RoomInfo roomInfo;
    int rating;
    String comment;
    Date date;

    public Review(UserData userData,  RoomInfo roomInfo, int rating, String comment, Date date) {
        this.userData = userData;
        this.roomInfo = roomInfo;
        this.rating = rating;
        this.comment = comment;
        this.date = date;
    }

    public UserData getUserData() {
        return userData;
    }

    public RoomInfo getRoomInfo() {
        return roomInfo;
    }

    public int getRating() {
        return rating;
    }

    @Override
    public int hashCode() {
        return this.getRoomInfo().hashCode();
    }

    @Override
    public String toString() {
        return STR."Review{userData=\{userData}, roomInfo=\{roomInfo.getRoomId()} - - - \{roomInfo.getRoomName()}, rating=\{rating}, comment='\{comment}\{'\''}\{'}'}";
    }

}

File: app/src/main/java/com/example/distributed_hotel_booking/data/Review.kt
package com.example.distributed_hotel_booking.data

import java.time.LocalDateTime
import java.time.ZoneId
import java.util.Date

data class Review(
    var userData: UserData = UserData(),
    val rating: Int,
    val comment: String,
    var roomInfo: RoomInfo = RoomInfo(),
    val date: Date = Date.from(LocalDateTime.now().atZone(ZoneId.systemDefault()).toInstant())
)